### PR TITLE
Port some other wanted modules to 3.0.

### LINKED ISSUE #193: m_hideidle for insp3
Hello, 

I'd like to see m_hideidle forwarded into extras for 3.0 -- it's a handy privacy feature.

Cheers,
g.

### PR DESCRIPTION
**Pull request by genius3000**
Porting some other wanted/requested modules to 3.0.
Please comment with any others and I'll add it to the list.

### List of modules:
- [x] m_antirandom - @SaberUK ported
- [x] m_conn_banner
- [x] m_hideidle
- [x] m_opmoderated
- [x] m_privdeaf - inspircd/inspircd#1603 to merge into m_deaf

Resolves #193.

### CODE REVIEW

File: 3.0/m_opmoderated.cpp [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+			return MOD_RES_PASSTHRU;
+
+		Channel* const chan = target.Get<Channel>();
+		if (CheckExemption::Call(exemptionprov, user, chan, "opmoderated") == MOD_RES_ALLOW)
+			return MOD_RES_PASSTHRU;
+
+		if (!chan->GetExtBanStatus(user, 'u').check(!chan->IsModeSet(&opmod)) && chan->GetPrefixValue(user) < VOICE_VALUE)
+		{
+			MessageTarget newtarget = target;
+			newtarget.status = '@';
```
Comment (genius3000) [inline 3.0/m_opmoderated.cpp]: Not sure if there's a better way to do this, without skipping the two On* calls below and sending to remote servers like timedbans does.

**Comment by Robby-** [comment]
About `m_privdeaf`, maybe it makes more sense to merge this into `m_deaf` instead of keeping this as an extras module? Similar to what happened with `m_noctcp_user`, it got merged into `m_noctcp`.

File: 3.0/m_conn_banner.cpp [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	}
+
+	void ReadConfig(ConfigStatus&) CXX11_OVERRIDE
+	{
+		text = ServerInstance->Config->ConfValue("connbanner")->getString("text");
+	}
+
+	void OnSetUserIP(LocalUser* user) CXX11_OVERRIDE
+	{
+		if (init.get(user))
```
Comment (SaberUK) [inline 3.0/m_conn_banner.cpp]: You unset this on `OnUserConnect` but there is no guarantee that a user's IP will not change after registration (e.g. via the IRCv3 resume spec).

File: 3.0/m_opmoderated.cpp [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+	}
+
+	void On005Numeric(std::map<std::string, std::string>& tokens) CXX11_OVERRIDE
+	{
+		tokens["EXTBAN"].push_back('u');
+	}
+
+	Version GetVersion() CXX11_OVERRIDE
+	{
+		return Version("Implements opmoderated channel mode +U (non-voiced messages sent to ops) and extban 'u'", VF_OPTCOMMON|VF_VENDOR);
```
Comment (SaberUK) [inline 3.0/m_opmoderated.cpp]:
```suggestion
		return Version("Implements opmoderated channel mode +U (non-voiced messages sent to ops) and extban 'u'", VF_OPTCOMMON);
```

File: 3.0/m_opmoderated.cpp (hunk 1 above)
Comment (SaberUK) [reply 3.0/m_opmoderated.cpp]: You should be able to update the status in the existing message target.

Alternately, there's a CUList in MessageDetails that you could add all unprivileged users to.

File: 3.0/m_opmoderated.cpp (hunk 3 above)
Comment (genius3000) [reply 3.0/m_opmoderated.cpp]: Will that cause a linking issue with 2.0's version and/or should I update that also?

Comment (SaberUK) [reply 3.0/m_opmoderated.cpp]: It won't cause linking issues but that should be updated too.

File: 3.0/m_conn_banner.cpp (hunk 2 above)
Comment (genius3000) [reply 3.0/m_conn_banner.cpp]: I'm starting to like the thought of a `OnUserPostInit` or `OnUserPreRegister` :smile:
I'd rather not leave ExtItems if I don't need to so if I check the registration state is none as well as the ExtItem it should cover all. The ExtItem will remain until connected and by then the registration state is definitely not none.

**Pull request merged**